storage/database/leveldb_database.go

```diff
@@ -472,18 +472,29 @@ func (db *levelDB) NewBatch() Batch {
 	return &ldbBatch{b: new(leveldb.Batch), ldb: db}
 }
 
+// ldbBatch is a write-only leveldb batch that commits changes to its host database
+// when Write is called. A batch cannot be used concurrently.
 type ldbBatch struct {
 	b    *leveldb.Batch
 	ldb  *levelDB
 	size int
 }
 
+// Put inserts the given value into the batch for later committing.
 func (b *ldbBatch) Put(key, value []byte) error {
 	b.b.Put(key, value)
 	b.size += len(value)
 	return nil
 }
 
+// Delete inserts the a key removal into the batch for later committing.
+func (b *ldbBatch) Delete(key []byte) error {
+	b.b.Delete(key)
+	b.size++
```

Comment:
May I ask why the size is increased?

Reply:
I followed the convention used in go-ethereum, however, this increase the size of batch since it does not delete the item instantly. As given key will be registered to the batch, not removing the item from the batch directly, I think this is the reason why they increase the size of the batch for `Delete` operation.

Reply:
Thank you :) So, it may not skip if any `Delete` operation exist in ldbBatch.